### DISCUSSION TITLE
Persist toggleable

### DISCUSSION THREAD

**Discussion by Terumi** [filamentphp/filament#9651] [Help]
Hello, 
It is a way to persist the toggled state of a column?
Some of my users want to see their data in a different way that the others.

**Comment by bigben3333** [comment]
up

**Comment by RoodFruit** [comment]
Very much needed. This will make sure different users can have different columns visible without every time making those changes!

**Comment by acornforth** [comment]
I've just been looking into this myself. It would be nice to have the ability to persist the toggle state of columns in the session. However in my case, which seems the same as OP. If this is to be a user preference, it will need to persist beyond the session and stored in user prefs somewhere. i expect this to be out-of-scope for filament, so I will be implementing something myself for this.

**Comment by acornforth** [comment]
Correction to my previous post, on testing myself, instead of feedback from testers, it _does_ persist in session already. just not across sessions

**Comment by Max-Lovell** [comment]
Any solution for across session persistance on column toggle yet?

**Reply by arnoldski** [reply]
You can use the "->persistFiltersInSession();", see https://filamentphp.com/docs/3.x/tables/filters/getting-started#persist-filters-in-session.

**Comment by memdufaizan** [comment]
Try this one
https://github.com/memdufaizan/filament-table-columns-persist

**Comment by alexmanase** [comment]
It seems to be a bug in Filament. I created an issue for that: https://github.com/filamentphp/filament/issues/18285
